backend/collab-service/spec/socket.spec.js

```diff
@@ -0,0 +1,64 @@
+const { createServer } = require("node:http");
```

Comment:
Should there be unit tests testing the service's response to certain events? It seems to just be testing the socket.io instead of any specific functionality of the service

Reply:
That's the issue with socket io and unit testing, I was not able to find an equivalent library to supertest which is what the others used to run the index.js essentially.

Reply:
In that case I don't see the point of adding these test cases, as they essentially test just the library

Reply:
Yeah I suppose that's fair. I'll remove along with the other changes requested.